its_hub/algorithms/planning_wrapper.py

```diff
@@ -0,0 +1,303 @@
+            approach_prompt = ApproachPromptTemplate.create_approach_prompt(prompt, approach)
+            
+            # Run base algorithm for this approach
+            approach_result = self.base_algorithm.infer(
+                lm, approach_prompt, approach_budget, return_response_only=False
+            )
+            
+            # Store approach-specific result
+            approach_results[approach] = approach_result
+            
+            # Collect responses for overall analysis
+            if hasattr(approach_result, 'responses'):
+                # For algorithms like BestOfN, SelfConsistency
+                combined_responses.extend(approach_result.responses)
+            elif hasattr(approach_result, 'all_responses'):
+                # For algorithms like ParticleFiltering
+                combined_responses.extend(approach_result.all_responses)
+            elif hasattr(approach_result, 'response_lists'):
+                # For algorithms like BeamSearch
+                for response_list in approach_result.response_lists:
+                    combined_responses.extend(response_list)
+            else:
+                # Fallback: treat as single response
+                combined_responses.append(str(approach_result.the_one))
+        
+        # Step 5: Select best approach based on algorithm-specific criteria
+        best_approach, best_result = self._select_best_approach(approach_results)
+        
+        # Create result object
+        result = PlanningWrappedResult(
+            plan=plan,
+            approaches=approaches,
+            approach_results=approach_results,
+            approach_budgets=approach_budgets,
+            combined_responses=combined_responses,
+            best_approach=best_approach,
+            best_approach_result=best_result,
+        )
+        
+        return result.the_one if return_response_only else result
+    
+    def _select_best_approach(
+        self, approach_results: Dict[str, AbstractScalingResult]
+    ) -> tuple[str, AbstractScalingResult]:
+        """Select the best approach based on algorithm-specific criteria."""
+        
+        # Default: select based on highest confidence/score if available
+        best_approach = None
+        best_result = None
+        best_score = float('-inf')
+        
+        for approach, result in approach_results.items():
+            score = self._get_result_score(result)
+            
+            if score > best_score:
+                best_score = score
+                best_approach = approach
+                best_result = result
+        
+        # Fallback to first approach if no scoring available
+        if best_approach is None:
+            best_approach = list(approach_results.keys())[0]
+            best_result = approach_results[best_approach]
+        
+        return best_approach, best_result
+    
+    def _get_result_score(self, result: AbstractScalingResult) -> float:
```

Comment:
if this turns out to be a useful thing, maybe we should implement it in subclasses of `AbstractScalingResult`.